### PR TITLE
Bind 2 high visibility PrintCore APIs

### PR DESCRIPTION
**Pull request by chamons**
- https://github.com/xamarin/xamarin-macios/issues/4713

### CODE REVIEW

**Comment by chamons** [comment]
There are a number of APIs that are possible to bind, but these were the most common/high visibility and easiest to test.

Tested with this snipped of manual code:

```
			var code = PMServer.CreatePrinterList (out PMPrinter[] printerList);
			if (code != PMStatusCode.Ok)
				throw new System.InvalidOperationException ();
			System.Console.WriteLine (printerList[0].ID);
			System.Console.WriteLine (printerList[0].HostName);
			System.GC.Collect (2);
			System.GC.Collect (2);
```

File: src/PrintCore/PrintCore.cs [hunk 1]
```diff
@@ -753,6 +753,26 @@ public PMStatusCode SetDefault ()
 			}
 		}
 
+		[DllImport (Constants.PrintCoreLibrary)]
+		extern static IntPtr PMPrinterGetID (IntPtr printer);
+
+		public string ID {
+			get {
+				return CFString.FetchString (PMPrinterGetID (handle));
```
Comment (chamons) [inline src/PrintCore/PrintCore.cs]: Not , true as https://developer.apple.com/documentation/applicationservices/1459606-pmprintergetid?language=objc has:

```You should not release the string without first retaining it.```

File: src/PrintCore/PrintCore.cs [hunk 2]
```diff
@@ -753,6 +753,26 @@ public PMStatusCode SetDefault ()
...
+
+		[DllImport (Constants.PrintCoreLibrary)]
+		extern static PMStatusCode PMPrinterCopyHostName (IntPtr printer, out IntPtr hostName);
+
+		public string HostName {
+			get {
+				PMStatusCode code = PMPrinterCopyHostName (handle, out IntPtr hostName);
+				if (code != PMStatusCode.Ok)
+					return null;
+				return CFString.FetchString (hostName, true);
```
Comment (chamons) [inline src/PrintCore/PrintCore.cs]: Releases as https://developer.apple.com/documentation/applicationservices/1462076-pmprintercopyhostname says:

```You are responsible for releasing the string.```

**Comment by chamons** [comment]
Forgot the api diff removal. 🤦‍♂️

**Comment by chamons** [comment]
Thoughts @spouliot ?

File: src/PrintCore/PrintCore.cs (hunk 1 above)
Comment (spouliot) [reply src/PrintCore/PrintCore.cs]: if you need to comment your own PR, before reviews, then it's a good sign those should be comment in the code ;-)

File: src/PrintCore/PrintCore.cs (hunk 2 above)
Comment (spouliot) [reply src/PrintCore/PrintCore.cs]: same

File: src/PrintCore/PrintCore.cs [hunk 3]
```diff
@@ -753,6 +753,26 @@ public PMStatusCode SetDefault ()
 			}
 		}
 
+		[DllImport (Constants.PrintCoreLibrary)]
+		extern static IntPtr PMPrinterGetID (IntPtr printer);
+
+		public string ID {
```
Comment (spouliot) [inline src/PrintCore/PrintCore.cs]: `Id` -> it's not an acronym

**Pull request merged**